### PR TITLE
Updated store providers

### PR DESCRIPTION
**Pull request by jubalm**
Updates here are part of a larger PR and contains only functionality. The UI is on another PR. There is a demo page in `#test` route to test the new store.

the Demo page will be removed before merging this PR.

### CODE REVIEW

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
I'll give this another look after the base branch issue has been resolved, just request another review.

File: app/ts/components/App.tsx
Comment (jubalm) [reply app/ts/components/App.tsx]: this PR depends on that branch #94 so i intentionally pointed it there until that one gets merged so i don't accidentally push to master.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Page doesn't load at all if you have no wallet installed:
> Uncaught Error: Wallet not installed or incompatible.

File: app/ts/components/Demo.tsx
Comment (KillariDev) [inline app/ts/components/Demo.tsx:128]: do we check that the tokens are valid? eg, address is a proper address? We could also check that the token exists, decimals/symbols could also be fetched from chain

File: app/ts/components/Demo.tsx
Comment (KillariDev) [inline app/ts/components/Demo.tsx:113]: should we have icon too?

File: app/ts/store/tokens.ts
Comment (KillariDev) [inline app/ts/store/tokens.ts:14]: I think this should be properly parsed and not casted

File: app/ts/store/tokens.ts
Comment (KillariDev) [inline app/ts/store/tokens.ts:16]:
```suggestion
		throw new Error('Token cache is corrupted.')
```

File: app/ts/store/tokens.ts
Comment (KillariDev) [inline app/ts/store/tokens.ts:57]: I think it would be better if these are bigints when parsed from storage and then stored as strings to storage

File: app/ts/store/tokens.ts
Comment (KillariDev) [inline app/ts/store/tokens.ts:66]:
```suggestion
		name: 'Rai Reflex Index',
```

File: app/ts/components/Demo.tsx
Comment (jubalm) [reply app/ts/components/Demo.tsx]: there is a separate component later to handle which icon to load. they will basically be basically be images files named from from addresses.

File: app/ts/components/Demo.tsx
Comment (jubalm) [reply app/ts/components/Demo.tsx]: i updated the demo for the add token, including pulling information from the Contract instance and calling their actual meta.

File: app/ts/store/tokens.ts
Comment (jubalm) [reply app/ts/store/tokens.ts]: i now updated the function to validate parsed json from cache including assertions to narrow down the resulting array

**Comment by jubalm** [comment]
> Page doesn't load at all if you have no wallet installed:
> 
> > Uncaught Error: Wallet not installed or incompatible.

There is a missing condition in checking account state listener which should be fixed in https://github.com/DarkFlorist/lunaria/pull/95/commits/5e15f46f74d76b384fb89d52e9f38613e02a31ff

File: app/ts/store/tokens.ts
Comment (jubalm) [reply app/ts/store/tokens.ts:89]: this change may break so i created #102 to address later

File: app/ts/components/Demo.tsx
Comment (jubalm) [reply app/ts/components/Demo.tsx]: created a new PR for the icon strategy #103

File: app/ts/components/Demo.tsx
Comment (MicahZoltu) [inline app/ts/components/Demo.tsx:39]: I currently just see `error` after clicking connect.  No details provided in the log or on the page.  Perhaps here we could just do:
```suggestion
		case 'rejected':
			return <>error: ${address.value.error.message}</>
```

**Review by MicahZoltu — DISMISSED** [review]
Getting closer to working (page now loads) but I cannot connect with interceptor attached (no MetaMask).

File: app/ts/components/Demo.tsx
Comment (jubalm) [reply app/ts/components/Demo.tsx]: looks like `window.ethereum` does not have a `addListener` function if you are using other than MetaMask to connect. I've changed it to `on` as I found that Brave also supports this.

FWIW: the function seem to vary between vendor and I doesn't seem to be on 1193 specs

File: app/ts/store/provider.ts
Comment (0xjimmy) [inline app/ts/store/provider.ts:45]: When network changes you don't need to reinitialize the provider, for a future optimization you could just make sure to remove event listener on disconnect instead.
Right now you, set the value of the provider store again, which triggers Signal effect in `app/ts/store/network.ts` which then queries the network from the provider.
You could just use the network provided in the network change event, this would remove the need for a signalEffect and rpc calls to fetch network.

**Review by 0xjimmy — APPROVED** [review]

File: app/ts/store/provider.ts
Comment (jubalm) [reply app/ts/store/provider.ts:45]: Will take a look closer as a separate issue #106

**Pull request merged**
